app/controllers/api/v2/home_controller.rb

```diff
@@ -11,9 +13,11 @@ def index
         Apipie.reload_documentation if Apipie.configuration.reload_controllers?
       end
 
+      add_smart_proxy_filters :status
       api :GET, "/status/", N_("Show status")
 
       def status
+        @remote_ip = request.remote_ip if detected_proxy
```

Comment:
why we'd add that only in case the proxy auth is used? I'd say it's reasonable to always tell which IP asked for the status

Reply:
My initial reasoning was that this was a way to see if authentication worked. Perhaps that's something we can rework and make it into a proper object with a link to the Smart Proxies API object page (something like `/api/smart_proxies/:id` IIRC).

Reply:
I'll take a look.

Reply:
I further considered this. I'd say that it's a potential information leak if we return it for all users. If a setup is misconfigured, it could leak data an attacker could use.

For now I'd prefer to get this in as it is. We can later add the Smart Proxy details. If we have this in Foreman 3.5 we can use it in scripts to verify things so I'd also ask about a cherry pick into 3.5.

Reply:
> Sorry for late response here, I wouldn't have problem merging as is (right after testing), if you wish to still get it to 3.5, ping me directly pls.

As discussed offline: let's postpone this to later and discuss the implications further.

> I was curious, what information an attacker gets if we add remote ip? Who's IP would it be?

If there's a reverse proxy in between that may be untrusted you could leak details. Defense in depth suggests you should leak as little information as possible to give attackers minimal information.

Reply:
> I further considered this. I'd say that it's a potential information leak if we return it for all users. If a setup is misconfigured, it could leak data an attacker could use.
> 
> For now I'd prefer to get this in as it is. We can later add the Smart Proxy details. If we have this in Foreman 3.5 we can use it in scripts to verify things so I'd also ask about a cherry pick into 3.5.

If an attacker has access to this data, it would be possible to run REX jobs; access the DB; see the logs and audits. I don't think this should block us from adding such a helpful information.